lib/SILGen/SILGenStmt.cpp

```diff
@@ -74,7 +74,9 @@ void SILGenFunction::eraseBasicBlock(SILBasicBlock *block) {
   assert(block->pred_empty() && "erasing block with predecessors");
   assert(block->empty() && "erasing block with content");
   if (block == StartOfPostmatter) {
-    StartOfPostmatter = &*std::next(block->getIterator());
+    StartOfPostmatter = (std::next(block->getIterator()) == F.end())
+                            ? nullptr
+                            : &*std::next(block->getIterator());
```

Comment:
Yeah, that seems reasonable. Although in this case, a `next_or_end` function probably makes more sense?